mongo/job_store.go

```diff
@@ -133,6 +133,40 @@ func (m *JobStore) CreateTask(ctx context.Context, jobID string, taskName string
 	return newTask, err
 }
 
+// GetTask retrieves the details of a particular task, from the collection, specified by its task name and associated job id
+func (m *JobStore) GetTask(ctx context.Context, jobID string, taskName string) (models.Task, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+	log.Info(ctx, "getting task from mongo DB", log.Data{"jobID": jobID, "taskName": taskName})
+
+	// If an empty jobID or taskName was passed in, return an error with a message.
+	if jobID == "" {
+		return models.Task{}, ErrEmptyIDProvided
+	} else if taskName == "" {
+		return models.Task{}, ErrEmptyTaskNameProvided
+	}
+
+	var job models.Job
+	job, err := m.findJob(s, jobID, job)
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return models.Task{}, ErrJobNotFound
+		}
+		return models.Task{}, err
+	}
+
+	var task models.Task
```

Comment:
If you move this to line 141, the variable can be used across the function, see below:

You can then do a search and replace of `models.Task{}` => `task`

See next comment for example of this.